### PR TITLE
Doc: Fixed typos

### LINKED ISSUE #6: Typos and doc corrections
We need to fix typos and doc corrections (grammar corrections)

### PR DESCRIPTION
**Pull request by tanryberdi**
- Fixed typos
- change ioutil.ReadAll to io.ReadAll
- Fixed #6

### CODE REVIEW

File: mux_httpserver_test.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func TestSchemeMatchers(t *testing.T) {
 		if resp.StatusCode != 200 {
 			t.Fatalf("expected a status code of 200, got %v", resp.StatusCode)
 		}
-		body, err := ioutil.ReadAll(resp.Body)
+		body, err := io.ReadAll(resp.Body)
```
Comment (vaguecoder) [inline mux_httpserver_test.go:31]: `ioutil` -> `io` is a valid change but let's not do it here. We can have another issue+PR for the same. Documentation PRs often don't touch the code parts.

File: README.md [hunk 2]
```diff
@@ -17,10 +17,10 @@ their respective handler.
 
 The name mux stands for "HTTP request multiplexer". Like the standard `http.ServeMux`, `mux.Router` matches incoming requests against a list of registered routes and calls a handler for the route that matches the URL or other conditions. The main features are:
 
-* It implements the `http.Handler` interface so it is compatible with the standard `http.ServeMux`.
+* It implements the `http.Handler` interface so, it is compatible with the standard `http.ServeMux`.
 * Requests can be matched based on URL host, path, path prefix, schemes, header and query values, HTTP methods or using custom matchers.
 * URL hosts, paths and query values can have variables with an optional regular expression.
-* Registered URLs can be built, or "reversed", which helps maintaining references to resources.
+* Registered URLs can be built, or "reversed", which helps to maintaining references to resources.
```
Comment (vaguecoder) [inline README.md:23]: I believe it should be either `to maintain` or `in maintaining`. Please check.

File: mux_httpserver_test.go (hunk 1 above)
Comment (vaguecoder) [reply mux_httpserver_test.go:31]: FYI #17 is created for the same

Comment (tanryberdi) [reply mux_httpserver_test.go]: done

File: README.md (hunk 2 above)
Comment (tanryberdi) [reply README.md]: done

**Comment by tanryberdi** [comment]
@vaguecoder please take a look 🙏

**Comment by tanryberdi** [comment]
rebased from `master`

**Review by vaguecoder — APPROVED** [review]
LGTM

**Comment by vaguecoder** [comment]
Thanks @tanryberdi Good job

**Pull request merged**